Remove release archive from static NAV site

lunkwill42 opened the pull request:
Manually maintaining a "mirror" of all releases on the NAV landing site makes no sense.  We've been using GitHub for years now, and we use the release mechanisms there to provide this functionality in a much more organized and direct fashion.

This PR removes the release archive from the static site and points to GitHub instead.  Also, since the front page references the latest release version number (which it used to get by looking at the content of the internal release archive), this updates the template to actually fetch it at build time, directly from GitHub's API.

lunkwill42 committed:
Point to GitHub releases as our release archive

There's no need to maintain this static archive of pointers to what is
already presented just fine on GitHub itself.

lunkwill42 committed:
Provide a link to GitHub releases

For those who have maybe kept a bookmark to this release page, this
update will tell them where to go instead.

lunkwill42 committed:
Fetch the latest release version from GitHub

We want to remove the local release archive, so let's fetch the latest
release version number directly from GitHub's API instead, at build
time.

lunkwill42 committed:
Drop all content from obsolete release archive

podliashanyk reviewed (commented):
Good idea to switch to GitHub releases instead.

I am getting a `404 Not Found` if I bookmark and go to `/releases/` on this branch, ref commit 23d4cbeb11f19273034213093c11855f8e0567cd

<img width="1512" alt="Screenshot 2024-05-22 at 10 14 23" src="https://github.com/Uninett/nav-landing-page/assets/60876078/f16f3fbe-1f34-4b3d-98ab-1a3f62a90cfe">

lunkwill42 committed:
Reinstate release archive index file

At least one file is required in the content section of the old release
archive in order for the /releases/ URL to still work (and show the link
to the GitHub release archive)

lunkwill42 commented:
> Good idea to switch to GitHub releases instead.
> 
> I am getting a `404 Not Found` if I bookmark and go to `/releases/` on this branch, ref commit [23d4cbe](https://github.com/Uninett/nav-landing-page/commit/23d4cbeb11f19273034213093c11855f8e0567cd)

Nice catch - This was caused by the last commit (6c02152), without me noticing.  Removing all the release content files also removed the releases index page.  I added back an empty file into to the content directory, and now it seems to work as intended.
